.github/workflows/publish-to-crates-io.yml

```diff
@@ -0,0 +1,23 @@
+# The purpose of this workflow is to publish the workspace's crates
+# whenever a tag is created. This baiscally boils down to running
+# `scripts/publish.rs` at the right time.
+
+name: "Publish to crates.io"
+
+on:
+  push:
+    tags:
+    - 'v*'
+
+jobs:
+  publish:
+    if: github.repository == 'bytecodealliance/cargo-component'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: rustup update stable && rustup default stable
+    - run: |
+        rustc ci/publish.rs
+        ./publish publish
+      env:
+        CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

Comment:
the github ui syntax coloring makes me think that maybe it is safer to put `env:` first and then `run: |`

Reply:
GitHub syntax coloring can be a little wonky at times, but ultimately this shouldn't matter.

For context, this CI job definition is a copy from other repros where it works as intended.